security/authorization/engine/engine.go

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package engine
+
+import (
+	pb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+	cel "github.com/google/cel-go/cel"
+	types "github.com/google/cel-go/common/types"
+	interpreter "github.com/google/cel-go/interpreter"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+// AuthorizationArgs is the input of the CEL-based authorization engine.
+type AuthorizationArgs struct {
+	md       metadata.MD
+	peerInfo *peer.Peer
+}
+
+// ActivationImpl is an implementation of interpreter.Activation.
```

Comment:
If this type only exists to satisfy an interface, do not export it, and return an interpreter.Activation from a `NewActivation()` constructor.